lnwallet/reservation.go

```diff
@@ -296,6 +296,15 @@ func (r *ChannelReservation) CommitConstraints(csvDelay, maxHtlcs uint16,
 		return ErrCsvDelayTooLarge(csvDelay, maxDelay)
 	}
 
+	// The channel reserve should always be greater or equal to our dust
+	// limit. The reservation request should be denied if otherwise.
+	//
+	// TODO: change this if we allow to manually specify the dust limit.
+	dustLimit := DefaultDustLimit()
+	if chanReserve < dustLimit {
+		return ErrChanReserveTooSmall(chanReserve, dustLimit)
+	}
```

Comment:
We also need to ensure that the reserve we propose is above the dust limit. If it is, then we should increase the reserve until it's safely above the dust limit.

Reply:
Fixed.